src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -115,6 +124,10 @@ internal void Dispose(bool joinThreads)
             {
                 // set _disposing so that we don't attempt restart the host anymore:
                 _disposing = true;
+                if (Interlocked.Exchange(ref _processExitHandling, ProcessExitHandled) == ProcessExitHooked)
+                {
+                    Process.Exited -= ProcessExitedHandler;
+                }
```

Comment:
For maintain some consistency I'd also signal the semaphore here (call ```_disposeSemaphore.Release()```) and add a comment on that semaphore that it is signaled when the ProcessExited handler is removed and can no longer be executed.